src/tests/conformance/invalid_message/disconnect.rs

```diff
@@ -0,0 +1,303 @@
+                message.short_string()
+            );
+        }
+        Err(err) => {
+            panic!("Error waiting for Ping: {:?}", err);
+        }
+    }
+
+    // Use Ping-Pong to check node's response.
+    // We expect a disconnect.
+    match synthetic_node
+        .ping_pong_timeout(node.addr(), DC_TIMEOUT)
+        .await
+    {
+        Err(PingPongError::ConnectionAborted) => {}
+        Ok(_) => panic!("Message was ignored."),
+        Err(err) => panic!("Connection was not aborted: {:?}", err),
+    }
+
+    synthetic_node.shut_down();
+    node.stop().unwrap();
+}
+
+#[tokio::test]
+async fn get_data_with_mixed_types() {
+    // zcashd: fail (replies with Block)
+    // zebra:  pass
+    let genesis_block = Block::testnet_genesis();
+    let mixed_inv = vec![genesis_block.inv_hash(), genesis_block.txs[0].inv_hash()];
+    let message = Message::GetData(Inv::new(mixed_inv));
+    run_test_case_message(message).await.unwrap();
+}
+
+#[tokio::test]
+async fn inv_with_mixed_types() {
+    // zcashd: fail (message ignored)
+    // zebra:  pass
+
+    // Inv with mixed inventory (using non-genesis block since all node's "should" have genesis already,
+    // which makes advertising it non-sensical).
+    let block_1 = Block::testnet_1();
+    let mixed_inv = vec![block_1.inv_hash(), block_1.txs[0].inv_hash()];
+    let message = Message::Inv(Inv::new(mixed_inv));
+    run_test_case_message(message).await.unwrap();
+}
+
+#[tokio::test]
+async fn addr_without_timestamp() {
+    // zcashd: pass
+    // zebra:  fail (replies with Reject(Malformed))
+
+    // Encode a custom type which mimics Message::Addr but without the timestamp.
+    let bytes = AddrWithoutTimestamp::new().encode().unwrap();
+
+    run_test_case_bytes(bytes).await.unwrap();
+}
+
+/// Mimics the encoding of [`NetworkAddr`] but excludes the timestamp field.
+struct NetAddrWithoutTimestamp(NetworkAddr);
+impl NetAddrWithoutTimestamp {
+    fn new() -> Self {
+        Self(NetworkAddr::new(SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+            0,
+        )))
+    }
+}
+impl Codec for NetAddrWithoutTimestamp {
+    fn encode(&self, buffer: &mut Vec<u8>) -> io::Result<()> {
+        self.0.encode_without_timestamp(buffer)
+    }
+
+    fn decode(_bytes: &mut io::Cursor<&[u8]>) -> io::Result<Self>
+    where
+        Self: Sized,
+    {
+        unimplemented!("This is unused");
+    }
+}
+
+/// Mimics the encoding of a broken [`Message::Addr`].
+/// Contains a single [`NetAddrWithoutTimestamp`].
+///
+/// This is used by the [`addr_without_timestamp()`] test. [`Message::Addr`]
+/// cannot be used for this, as it does not support encoding without a timestamp (as
+/// this is obsolete behaviour).
+struct AddrWithoutTimestamp(Vec<NetAddrWithoutTimestamp>);
+impl AddrWithoutTimestamp {
+    fn new() -> Self {
+        Self(vec![NetAddrWithoutTimestamp::new()])
+    }
+
+    fn encode(&self) -> io::Result<Vec<u8>> {
```

Comment:
I think this is correct - but I did spend a lot of time looking and re-doing this, so I might have blind-sight. Additional scrutiny would be appreciated.

Reply:
Another option for this is to use the `NetworkAddr` implementation and remove the range of bytes corresponding to the timestamp with e.g. `Vec::drain`.

Reply:
I think your idea is a lot cleaner. Will change to that.